crates/oxc_linter/src/rules/unicorn/prefer_query_selector.rs

```diff
@@ -0,0 +1,207 @@
+use lazy_static::lazy_static;
+use miette::diagnostic;
+use oxc_ast::{
+    ast::{Argument, Expression},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use std::collections::HashMap;
+
+use crate::{context::LintContext, rule::Rule, utils::is_dom_node_call, AstNode, Fix};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(prefer-query-selector): Prefer `.{0}()` over `.{1}()`.")]
+#[diagnostic(severity(Advice), help("It's better to use the same method to query DOM elements. This helps keep consistency and it lends itself to future improvements (e.g. more specific selectors)."))]
```

Comment:
Shall we change `Advice` back to `warning`?

Reply:
Yeah, I forgot about that.